packages/teleterm/src/ui/DocumentCluster/ClusterResources/Databases/useDatabases.ts

```diff
@@ -14,51 +14,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { AttemptStatus } from 'shared/hooks/useAsync';
-
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import { Database, ServerSideParams } from 'teleterm/services/tshd/types';
 import { routing } from 'teleterm/ui/uri';
-import { GatewayProtocol } from 'teleterm/ui/services/clusters';
+import { GatewayProtocol, makeDatabase } from 'teleterm/ui/services/clusters';
 
 import { useServerSideResources } from '../useServerSideResources';
-import { useClusterContext } from '../../clusterContext';
-
-function getEmptyTableText(status: AttemptStatus) {
-  switch (status) {
-    case 'error':
-      return 'Failed to fetch databases.';
-    case '':
-      return 'Searching…';
-    case 'processing':
-      return 'Searching…';
-    case 'success':
-      return 'No databases found.';
-  }
-}
 
 export function useDatabases() {
   const appContext = useAppContext();
-  const clusterContext = useClusterContext();
-  const documentUri = clusterContext.documentUri;
 
-  const {
-    fetchAttempt,
-    updateQuery,
-    agentFilter,
-    prevPage,
-    nextPage,
-    updateSearch,
-    onAgentLabelClick,
-    updateSort,
-    pageCount,
-  } = useServerSideResources<Database>(
-    { fieldName: 'name', dir: 'ASC' }, // default sort
-    (params: ServerSideParams) =>
-      appContext.resourcesService.fetchDatabases(params)
-  );
+  const { fetchAttempt, documentUri, ...serverSideResources } =
+    useServerSideResources<Database>(
+      { fieldName: 'name', dir: 'ASC' }, // default sort
+      (params: ServerSideParams) =>
+        appContext.resourcesService.fetchDatabases(params)
+    );
 
-  function connect(db: Database, dbUser: string): void {
+  function connect(db: ReturnType<typeof makeDatabase>, dbUser: string): void {
```

Comment:
Edit: Oh, I misunderstood what's happening here. I suppose for now it's a necessary evil.

Another minor nit but I'd probably use `Database` here. For me `ReturnType` seems like one of those utility types that should be reserved for functions that accept other functions. Most type systems that I've seen don't really use the concept of "I want to use the return type of this other function", I feel like TypeScript has it just because it needs to support all those dynamic JS acrobatics.